core/src/main/java/io/questdb/cairo/mig/EngineMigration.java

```diff
@@ -197,16 +193,16 @@ static void backupFile(FilesFacade ff, Path src, Path toTemp, String backupName,
         int copyPathLen = toTemp.length();
         try {
             toTemp.concat(backupName).put(".v").put(version);
+            int versionLen = toTemp.length();
             for (int i = 1; ff.exists(toTemp.$()); i++) {
                 // if backup file already exists
                 // add .<num> at the end until file name is unique
                 LOG.info().$("backup dest exists [to=").$(toTemp).I$();
-                toTemp.trimTo(copyPathLen);
-                toTemp.concat(backupName).put(".v").put(version).put(".").put(i);
+                toTemp.trimTo(versionLen).put('.').put(i);
             }
 
-            LOG.info().$("backing up [file=").$(src).$(",to=").$(toTemp).I$();
-            if (ff.copy(src.$(), toTemp.$()) < 0) {
+            LOG.info().$("backing up [file=").$(src).$(", to=").$(toTemp).I$();
+            if (ff.copy(src.$(), toTemp) < 0) {
```

Comment:
now this yet another path manipulation introduced a bug that need not to be there and is hard to fix. I would strongly encourage not to mess with path where it is not needed. 

This instance of path is not zero terminated now.